docs/develop.md

```diff
@@ -1,18 +1,426 @@
+
+# later commits can be pushed directly
+git push
+```
+
+Frequently committing and pushing your code helps to keep a clear project
+history and can save you from losing work. Also, others will be able to see what
+you're working on, which can prevent double work and can be useful to ask for
+confirmation or advise in an early stage, before you put in a lot of effort.
+
+### Open a (draft) pull request
+
+When you first push your changes to the remote and navigate to the repository on
+GitHub, you will see a popup that suggests to open a new pull request. If you
+click this, you will see a comparison between your branch and the 'target'. If
+you're working on a fork, you can set the target to the "upstream" repo, i.e.
+the original WAM2layers repository. Usually the `main` branch will be the
+target.
+
+Add a clear title and description (these are often edited later), pointing to
+the relevant issue(s) that the PR addresses. On the green button you can select
+"Open draft PR".
+
+When you open a pull request, some automatic checks are triggered. For example,
+a preview of the documentation is build based on the code in the pull request.
+You can look at the details of the checks to preview the documentation, or, if
+they fail, what causes the issue(s).
+
+### The review process
+
+When you are happy with your changes, you can mark the PR "ready for review".
+GitHub will automatically suggest some reviewers, or you can pick your own
+choice. Anyway, every PR must be reviewed and approved by a core developer
+before it is merged.
+
+Reviewing others' PRs can be a good way to contribute to the community. GitHub
+provides great functionality for commenting on individual lines of code, making
+suggestions, et cetera. Please be welcoming and constructive in your feedback. A
+good review involves verifying that the new code is clear, well documented, and
+does what it's supposed to do, testing that the code (still) works, checking
+that it conforms to the guidelines.
+
+When the reviewer is happy with the code, they can approve the changes. At that
+point, the code can be merged and becomes part of the "main" codebase. However,
+a new release is required for it to become part of the PyPI and Zenodo
+publications.
+
+### Making a release
+
+These instructions are intended for core developers.
+
+You need an [account on PyPI](https://pypi.org/account/register/) with owner
+rights on WAM2layers.
+
+- Create a new branch, e.g. `prep_release_vXXX`
+- Update "version" in pyproject.toml and in citation.cff (use [semantic
+  versioning](https://semver.org/))
+- Make sure the code is neatly formatted (see above)
+- Make sure the changelog is up to date; update the "Unreleased" header with the
+  new version, and add a new "Unreleased" header above
+- Review and merge the release branch
+- [Make a release on GitHub](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository)
+- Publish to PyPI:
+    ```
```

Comment:
not so clear how we go from the release on Github all of a sudden to this code on the command line below